src/python/pants/core/util_rules/system_binaries.py

```diff
@@ -58,8 +61,8 @@ def _fingerprint(content: bytes | bytearray | memoryview | None = None) -> str:
 
     @classmethod
     def fingerprinted(
-        cls, path: str, representative_content: bytes | bytearray | memoryview
-    ) -> BinaryPath:
+        cls: type[BinaryPathT], path: str, representative_content: bytes | bytearray | memoryview
+    ) -> BinaryPathT:
```

Comment:
As I understand how Python typing works (experiments in https://mypy-play.net/?mypy=1.1.1&python=3.9&gist=5c8085580869f7846e12cd77b7e93e6e), I think this can potentially skip the TypeVars:

```suggestion
        cls, path: str, representative_content: bytes | bytearray | memoryview
    ) -> typing_extensions.Self:
```

Even without `Self` (https://docs.python.org/3/library/typing.html#typing.Self), I don't think the overall class needs the `Generic[BinaryPathT]`: what's written for this class method works with just `class BinaryPath:`.

I guess the typevar might be necessary if subclasses of `BinaryPath` might need to be explicitly constructing a superclass, For instance, if `MySubPythonExecutable(PythonExecutable)` needs to have `fingerprinted() -> PythonExecutable`, rather than `fingerprinted() -> MySubPythonExecutable`. But that seems a bit unusual.

Reply:
Ah, yeah - I forgot about the backporting. Much nicer: #18763

Reply:
Oh yeah I forget too. I was like "if only we used Python 3.11"